src/Dfe.Spi.GraphQlApi.Infrastructure.RegistryApi/RegistryApiRegistryProvider.cs

```diff
@@ -13,17 +15,28 @@ namespace Dfe.Spi.GraphQlApi.Infrastructure.RegistryApi
     public class RegistryApiRegistryProvider : IRegistryProvider
     {
         private readonly IRestClient _restClient;
+        private readonly ISpiExecutionContextManager _executionContextManager;
         private readonly ILoggerWrapper _logger;
 
-        public RegistryApiRegistryProvider(IRestClient restClient, RegistryConfiguration configuration, ILoggerWrapper logger)
+        public RegistryApiRegistryProvider(
+            IRestClient restClient, 
+            RegistryConfiguration configuration, 
+            ISpiExecutionContextManager executionContextManager,
+            ILoggerWrapper logger)
         {
             _restClient = restClient;
+            _executionContextManager = executionContextManager;
             _restClient.BaseUrl = new Uri(configuration.RegistryApiBaseUrl, UriKind.Absolute);
             if (!string.IsNullOrEmpty(configuration.RegistryApiFunctionKey))
             {
                 _restClient.DefaultParameters.Add(new Parameter("x-functions-key", configuration.RegistryApiFunctionKey,
                     ParameterType.HttpHeader));
             }
+            if (!string.IsNullOrEmpty(configuration.RegistryApiSubscriptionKey))
+            {
+                _restClient.DefaultParameters.Add(new Parameter("Ocp-Apim-Subscription-Key", configuration.RegistryApiSubscriptionKey,
```

Comment:
Would be good to common-ise this header in the `Dfe.Spi.Http` lib - there's even a headers constant class in there.

Reply:
In fact, you don't even need to include it explicitly, as `Dfe.Spi.Http.Server` has it as a dependency.